src/main/java/nl/tudelft/cse1110/andy/result/ResultBuilder.java

```diff
@@ -259,15 +259,34 @@ public Result build() {
             GradeValues grades = GradeValues.fromResults(coverageResults, codeCheckResults, mutationResults, metaTestResults);
             GradeWeight weights = GradeWeight.fromConfig(ctx.getRunConfiguration().weights());
 
+            this.checkExternalProcessExit();
+
             int finalGrade = calculateFinalGrade(grades, weights);
 
-            ExternalProcess process = ctx.getExternalProcess();
-            if (!process.hasExitedNormally()) {
-                genericFailureMessage = String.format("External process crashed with exit code %d: %s",
-                        process.getExitCode(), process.getErrorMessages());
+            GenericFailure genericFailure = null;
+
```

Comment:
Can we simply call a "buildAnyPossibleFailure()"? This method would then create a GenericFailure if one needed to be created. This method would also call the checkExternalProcess thingy

Reply:
Thank you for the suggestion, I extracted a method that builds the generic failure using the class variables. However, there is currently no way to move the call to `checkExternalProcessExit` to this method, as `checkExternalProcessExit` is used to set a failure flag if the process has crashed, which is then used in `calculateFinalGrade`, so it has to be called before calculating the grade.

```java
this.checkExternalProcessExit();

int finalGrade = calculateFinalGrade(grades, weights);

GenericFailure genericFailure = this.buildGenericFailure();
```

Reply:
I refactored the way GenericFailure objects are built and stored within the ResultBuilder class, and now it is possible to move this method call.